### ISSUE TITLE
Area chart not displaying latest data on the chart according 

### ISSUE THREAD

**Issue by 99sdawkhar** [recharts/recharts#5196]
## Do you want to request a feature or report a bug?

This appears to be a bug. It functioned correctly in earlier versions of Recharts but stopped working as expected since version 2.

## What is the current behavior?

I encounter this issue when using the <ResponsiveContainer /> wrapper around an AreaChart. When plotting a large dataset, Recharts internally calculates an interval for displaying points. Although the interval prop works well for smaller datasets, with larger datasets (e.g., 500+ data points), Recharts displays only a subset of points, likely for performance reasons. This optimization skips intermediate points, but unfortunately, it omits the last point in the dataset, which is critical in our use case.

## Steps to reproduce

I have created a minimal reproducible demo with recharts examples but with larger dataset: [[Demo Link Here](https://codesandbox.io/p/sandbox/simple-area-chart-forked-2mjk35)]

## Expected behavior: 
The first and last points in the dataset should always display, regardless of interval adjustments or screen resizing, or there should be an option to allow/disallow this behavior. 
**For expexted behaviour in above example please change recharts version to 1.8.6 or lesser**
	
*Note: Please note that this use to work well in older versions i.e <= 1.8.6 but doesn't work in newer versions. Idk its thats how the library was updated.*

## Environment details

This issue appears in some Windows devices and can be reproduced on any browser by zooming to 110/125 %.

**Comment by ckifer** [comment]
Can you try setting the interval prop to "preserveStartEnd"?

**Comment by 99sdawkhar** [comment]
Yes @ckifer, I have tried all interval values on x-axis but that doesn't help.

**Comment by ckifer** [comment]
The breaking changes between 1.x and 2.x are difficult because the previous maintainers never documented what they were.

---

That being said, this looks like this displaying the last data point just fine 
<img width="751" alt="image" src="https://github.com/user-attachments/assets/fcb55d67-39f6-4f3b-b485-3d8e30bcd358">

**Comment by ckifer** [comment]
It looks like `interval="preserveEnd"` has the same result as no interval prop did in 1.8.6
<img width="749" alt="image" src="https://github.com/user-attachments/assets/625f9c09-f629-4712-a9ec-831233b5fdad">

**Comment by 99sdawkhar** [comment]
Apologies, @ckifer, I've updated the dataset to include 1221 points, aligning it with my actual dataset. Could you check it now?

To clarify, the issue isn’t with the x-axis. The problem is that the tooltip doesn’t display details for the last data point when hovering over it. You can replicate this behavior by zooming your browser to 110% or 125%.

**Comment by ckifer** [comment]
Oh ok that wasn't clear from the original issue. Will look at it again when I can.

**Comment by ckifer** [comment]
at 110% zoom I can get the last datapoint normally
<img width="152" alt="image" src="https://github.com/user-attachments/assets/782918a9-28c2-4500-b313-531645f44028">

at 120% and beyond I see what you mean.

I wasn't one of the original implementers of all of this so to be completely honest I don't know how this works. It looks like the index gets updated by larger numbers the smaller the chart is (or the more datapoints there are). This makes sense as its impossible to fire an event for each, there are just too many data points.

But I agree with you that the first and last datapoints should always be included and it seems that changed somehow. Labelled this as a bug but no promises on when it can be looked into more. 

---

Linking this issue cause its slightly related (but not really)
https://github.com/recharts/recharts/issues/4884

**Comment by PavelVanecek** [comment]
Tooltip looks just fine to me even on 400% zoom. Where should I look?

<img width="1649" height="1685" alt="Image" src="https://github.com/user-attachments/assets/92c2276b-e600-4266-b5c4-adb5b6bc021e" />

**Comment by PavelVanecek** [comment]
Same on 3.x.x

**Comment by ckifer** [comment]
yeah I think we can close this
